tests/framework/clients/corral/corral.go

```diff
@@ -30,24 +36,49 @@ func SetupCorralConfig(configVars map[string]string) error {
 	return nil
 }
 
+// SetCustomRepo sets a custom repo for corral to use. It takes a string as a parameter which is the repo you want to use
+func SetCustomRepo(repo string) error {
```

Comment:
I'm concerned about this function. I always foresaw the corral-packages already cloned and built in a test environment. I'm not a big fan of this being a function. Like are we always going to clone and rebuild the repo when testing in Jenkins?

Reply:
My understanding was that we would plan on publicly publishing the corral-packages repo at some point, but until then we need a solution that would still use a local build. Plus, this makes it easier to test corrals from other repos (i.e. your own fork) instead of relying on the package being in rancher's corral-packages repo.

Reply:
Ok that makes sense, and just saw there is a config file flag for it too.

Reply:
Caleb, I wanted to reopen this discussion as thought about this more. How necessary is that we would need to clone from repos from other forks? Back to your previous answer why we would need this function without publicly published corral-packages? Why couldn't we clone the repo beforehand in the JenkinsFile, or the DockerFile?